wezterm-gui/src/glyphcache.rs

```diff
@@ -4239,6 +4259,66 @@ impl<T: Texture2d> GlyphCache<T> {
                     );
                 }
             }
+            BlockKey::Braille(dots_pattern) => {
+                // `dots_pattern` is a byte whose bits corresponds to dots
+                // on a 2 by 4 dots-grid.
+                // The position of a dot for a bit position (1-indexed) is as follow:
+                // 1 4  |
+                // 2 5  |<- These 3 lines are filled first (for the first 64 symbols)
+                // 3 6  |
+                // 7 8  <- This last line is filled last (for the remaining 192 symbols)
+                //
+                // NOTE: for simplicity & performance reasons, a dot is a square not a circle.
+
+                let cell_width = self.metrics.cell_size.width as f32 / 2.;
```

Comment:
I'm finding this naming a bit confusing; `cell_width` is used in many places in wezterm to refer to the width of the terminal cell and I'd assumed that was true here and getting confused about how the dots fit in the cell, but then I saw that this is actually half that size!

Could you make a pass through to clarify these?

I think `square_length` is the key metric being computed here, so I'd suggest simplifying these into something like this:

```rust
   let square_width = self.metrics.cell_size.width as f32 / 4;
   let topleft_offset_x = self.metrics.cell_size.width as f32 / 8;
   let square_height = self.metrics.cell_size.height as f32 / 4;
   let topleft_offset_y = self.metrics.cell_size.height as f32 / 8;
```

I think this makes it a bit easier to visualize the positioning when thinking about the cell overall and how that gets divided up into an interior grid.

I think it's probably a good idea to have a separate `square_width` and `square_height` because different fonts have different aspect ratios, which makes it potentially problematic to use a metric derived from the X in the Y direction.

Reply:
Yeah I was thinking that `cell_width` isn't great naming either, sorry for the confusion :eyes:
As for the suggested vars definition, I find it way harder to see how it works overall..

I'd be ok to rename the badly named var to sth like `dot_area_width` maybe (or you have better naming?), but calculating the `topleft_offset_x` like you suggest isn't great IMO, because as soon as you want to have a slightly bigger square, you're out of luck, and have to rethink all your formulas. However doing this with my use of variables I believe is as easy as changing the `square_length` value, and the rest will adapt.
I don't think it's worth removing the intermediate variables, as in a release build llvm will probably find ways to optimize things.. And it's not as if the code was a hot-path for the rendering, it's only called a few times to build the cache.

If you really want I can do what you suggest (or you can change the code yourself afterward :eyes:), but know that I personally wouldn't want to maintain these simplified definitions if it was my project.

---

> I think it's probably a good idea to have a separate `square_width` and `square_height` because different fonts have different aspect ratios, which makes it potentially problematic to use a metric derived from the X in the Y direction.

I don't agree, if the dots were circles, would you like to have ovals? I don't think so..
And it's the same for sharp dots, I think these should be actual squares, not ~rectangles.

For the fonts with different aspect ratios, do you have something in mind that wouldn't look great? Maybe we can find an optimal `square_length` with some min/max of some other values?

Also, did you see my PR comment here: https://github.com/wez/wezterm/pull/919#issuecomment-873421351 ?